src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -5067,44 +5067,26 @@ public override IOperation VisitAnonymousObjectCreation(IAnonymousObjectCreation
-
             SpillEvalStack();
 
             var initializerBuilder = ArrayBuilder<IOperation>.GetInstance(operation.Initializers.Length);
             for (int i = 0; i < operation.Initializers.Length; i++)
             {
-                IOperation initializer = operation.Initializers[i];
-                IPropertySymbol initializedProperty = properties[i];
+                var simpleAssignment = (ISimpleAssignmentOperation)operation.Initializers[i];
+                var propertyReference = (IPropertyReferenceOperation)simpleAssignment.Target;
 
-                // Due to https://github.com/dotnet/roslyn/issues/25266,
-                // the initializer may or may not be a simple assignment to anonymous type property.
-                if (initializer.Kind == OperationKind.SimpleAssignment)
-                {
-                    var simpleAssignment = (ISimpleAssignmentOperation)initializer;
-                    if (simpleAssignment.Target.Kind == OperationKind.PropertyReference)
-                    {
-                        var propertyReference = (IPropertyReferenceOperation)simpleAssignment.Target;
-
-                        // Due to https://github.com/dotnet/roslyn/issues/22736,
-                        // Instance is null for property reference in an anonymous initializer.
-                        if (propertyReference.Instance == null &&
-                            propertyReference.Property == initializedProperty &&
-                            propertyReference.Arguments.IsEmpty)
-                        {
-                            IOperation visitedTarget = new PropertyReferenceExpression(initializedProperty, propertyReference.Instance, ImmutableArray<IArgumentOperation>.Empty,
-                                semanticModel: null, propertyReference.Syntax, propertyReference.Type, propertyReference.ConstantValue, IsImplicit(propertyReference));
-                            IOperation visitedValue = visitAndCaptureInitializer(initializedProperty, simpleAssignment.Value);
-                            var visitedAssignment = new SimpleAssignmentExpression(visitedTarget, isRef: simpleAssignment.IsRef, visitedValue,
-                                semanticModel: null, simpleAssignment.Syntax, simpleAssignment.Type, simpleAssignment.ConstantValue, IsImplicit(simpleAssignment));
-                            initializerBuilder.Add(visitedAssignment);
-                            continue;
-                        }
-                    }
-                }
+                Debug.Assert(propertyReference != null);
+                Debug.Assert(propertyReference.Arguments.IsEmpty);
+                Debug.Assert(propertyReference.Instance != null);
+                Debug.Assert(propertyReference.Instance.Kind == OperationKind.InstanceReference);
+                Debug.Assert(((IInstanceReferenceOperation)propertyReference.Instance).ReferenceKind == InstanceReferenceKind.ImplicitReceiver);
```

Comment:
>((IInstanceReferenceOperation)propertyReference.Instance).ReferenceKind == InstanceReferenceKind.ImplicitReceiver [](start = 29, length = 113)

Should probably assert this invariant in test visitors as well.

Reply:
Pretty much all these asserts should be there.

---
In reply to: [197207252](https://github.com/dotnet/roslyn/pull/28005#discussion_r197207252) [](ancestors = 197207252)